airflow/executors/mesos_executor.py

```diff
@@ -0,0 +1,422 @@
+        Subscribe to Mesos Master
+
+        :param driver: Mesos driver object
+        """
+        from airflow.models import Connection
+
+        # Update the Framework ID in the database.
+        conn_id = FRAMEWORK_CONNID_PREFIX + get_framework_name()
+        connection = session.query(Connection).filter_by(conn_id=conn_id).first()
+        if connection is None:
+            connection = Connection(conn_id=conn_id, conn_type='mesos_framework-id', extra=driver.frameworkId)
+        else:
+            connection.extra = driver.frameworkId
+
+        session.commit()
+        self.driver = driver
+
+    def status_update(self, update):
+        """Update the Status of the Tasks. Based by Mesos Events."""
+        task_id = update["status"]["task_id"]["value"]
+        task_state = update["status"]["state"]
+
+        self.log.info("Task %s is in state %s", task_id, task_state)
+
+        try:
+            key = self.task_key_map[task_id]
+
+        except KeyError:
+            # The map may not contain an item if the framework re-registered
+            # after a failover.
+            # Discard these tasks.
+            self.log.info("Unrecognised task key %s", task_id)
+            return
+
+        if task_state == "TASK_FINISHED":
+            self.result_queue.put((key, State.SUCCESS))
+            return
+
+        if task_state in ('TASK_LOST', 'TASK_KILLED', 'TASK_FAILED'):
+            self.result_queue.put((key, State.FAILED))
+            return
+
+
+class MesosExecutor(BaseExecutor):
+    """
+    MesosExecutor allows distributing the execution of task
+    instances to multiple mesos workers.
+
+    Apache Mesos is a distributed systems kernel which abstracts
+    CPU, memory, storage, and other compute resources away from
+    machines (physical or virtual), enabling fault-tolerant and
+    elastic distributed systems to easily be built and run effectively.
+    See http://mesos.apache.org/
+    """
+
+    class MesosFramework(threading.Thread):
+        """MesosFramework class to start the threading"""
+
+        def __init__(self, client):
+            threading.Thread.__init__(self)
+            self.client = client
+            self.stop = False
+
+        def run(self):
+            try:
+                self.client.register()
+            except KeyboardInterrupt:
+                print("Stop requested by user, stopping framework....")
+
+    def __init__(self):
+        super().__init__()
+        self.commands_to_run = []
+        self.task_queue = Queue()
+        self.result_queue = Queue()
+        self.driver = None
+        self.client = None
+        self.mesos_framework = None
+
+    @provide_session
+    def start(self, session=None):
+        """Setup and start routine to connect with the mesos master"""
+        if not conf.get('mesos', 'MASTER'):
+            self.log.error("Expecting mesos master URL for mesos executor")
+            raise AirflowException("mesos.master not provided for mesos executor")
```

Comment:
```suggestion
```

see previous comment and config loading

Reply:
Yeah, I remove all the check lines where I know we have ever a default value.